packages/fetchai/connections/soef/connection.py

```diff
@@ -91,106 +147,349 @@ def __init__(
         self.unique_page_address = None  # type: Optional[str]
         self.agent_location = None  # type: Optional[Location]
         self.in_queue = None  # type: Optional[asyncio.Queue]
+        self._executor_pool: Optional[ThreadPoolExecutor] = None
+        self.chain_identifier: str = chain_identifier or "fetchai"
 
-    def send(self, envelope: Envelope) -> None:
+    @staticmethod
+    def _is_compatible_query(query: Query) -> bool:
         """
-        Send message handler.
+        Check if a query is compatible with the soef.
```

Comment:
Add a comment in new line: "Each query must contain a distance constraint type."

Reply:
done